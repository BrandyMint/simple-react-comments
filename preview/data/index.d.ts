export declare const commentsData: {
    authorUrl: string;
    avatarUrl: string;
    createdAt: Date;
    fullName: string;
    text: string;
}[];
