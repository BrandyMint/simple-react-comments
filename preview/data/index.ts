export const commentsData = [
  {
    authorUrl: '#',
    avatarUrl:
      'https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438',
    createdAt: new Date(),
    fullName: 'Alexey Ryabov',
    text: '<h1>hello whats up</h1>',
  },
  {
    authorUrl: '/asdf',
    avatarUrl:
      'https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438',
    createdAt: new Date(),
    fullName: 'Alexey Ryabov',
    text: 'reactomments is awesome!',
  },
];
