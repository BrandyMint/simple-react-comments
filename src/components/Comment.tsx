import * as React from 'react';
import { CommentEntity } from '../models';
import Link from './Link';

export interface Props {
  comment: CommentEntity;
  reactRouter: boolean;
}

const Comment: React.StatelessComponent<Props> = ({ comment, reactRouter }) => {
  const { authorUrl, avatarUrl, text, createdAt, fullName } = comment;
  return (
    <div className="rc-comment">
      <Link reactRouter={reactRouter} href={authorUrl}>
        <div
          className="rc-comment__avatar"
          style={{
            backgroundImage: `url(${avatarUrl})`,
          }}
        />
      </Link>
      <div className="rc-comment__col-right">
        <Link reactRouter={reactRouter} href={authorUrl}>
          <div className="rc-comment__name">{fullName}</div>
        </Link>
        <div className="rc-comment__time">{createdAt.toLocaleDateString()}</div>
        <div className="rc-comment__content">{text}</div>
      </div>
    </div>
  );
};

export default Comment;
