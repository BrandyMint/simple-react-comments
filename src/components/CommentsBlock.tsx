import * as React from 'react';
import { CommentEntity, StylesEntity } from '../models';
import CommentForm from './CommentForm';
import CommentSignIn from './CommentSignIn';
import CommentsList from './CommentsList';

const defaultStyles = {
  btn: base => ({ ...base }),
  comment: base => ({ ...base }),
  textarea: base => ({ ...base }),
};

export const CBContext = React.createContext<StylesEntity>(defaultStyles);

export interface Props {
  comments: CommentEntity[];
  signinUrl: string;
  isLoggedIn?: boolean;
  reactRouter?: boolean;
  styles?: StylesEntity;
  onSubmit(text: string): void;
}

export interface State {
  styles: StylesEntity;
}

class CommentsBlock extends React.Component<Props, State> {
  public static defaultProps: Partial<Props> = {
    isLoggedIn: false,
    reactRouter: false,
    styles: defaultStyles,
  };

  public static getDerivedStateFromProps(props, state) {
    return {
      styles: Object.assign({}, state.styles, props.styles),
    };
  }

  constructor(props) {
    super(props);
    this.state = {
      styles: defaultStyles,
    };
  }

  public render() {
    const {
      comments,
      isLoggedIn,
      signinUrl,
      onSubmit,
      reactRouter,
    } = this.props;

    return (
      <CBContext.Provider value={this.state.styles}>
        <div>
          <CommentsList comments={comments} reactRouter={reactRouter} />
          {isLoggedIn ? (
            <CommentForm onSubmit={onSubmit} />
          ) : (
            <CommentSignIn href={signinUrl} reactRouter={reactRouter} />
          )}
        </div>
      </CBContext.Provider>
    );
  }
}

export default CommentsBlock;
