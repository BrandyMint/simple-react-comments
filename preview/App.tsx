import * as React from 'react';
import CommentsBlock from '../src/index';
import { commentsData } from './data/index';

const styles = {
  fontFamily: 'arial',
  margin: '0 auto',
  marginTop: '50px',
  width: '50%',
};

const cardStyles = {
  borderRadius: '10px',
  boxShadow: '0px 2px 10px rgba(0, 0, 0, 0.2)',
  padding: '30px',
};

export interface State {
  comments: any;
}

class App extends React.Component<{}, State> {
  constructor(props) {
    super(props);
    this.state = {
      comments: commentsData,
    };
  }

  public render() {
    return (
      <div style={styles}>
        <h1>react-simple-comments</h1>
        <div style={cardStyles}>
          <CommentsBlock
            comments={this.state.comments}
            signinUrl={'/signin'}
            isLoggedIn
            reactRouter
            onSubmit={text => {
              if (text.length > 0) {
                this.setState({
                  comments: [
                    ...this.state.comments,
                    {
                      authorUrl: '#',
                      avatarUrl:
                        'https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438',
                      createdAt: new Date(),
                      fullName: 'Alexey Ryabov',
                      text,
                    },
                  ],
                });
                console.log('submit:', text);
              }
            }}
          />
        </div>
      </div>
    );
  }
}

export default App;
