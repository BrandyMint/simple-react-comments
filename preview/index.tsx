import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { BrowserRouter, Route } from 'react-router-dom';
import './css/style.scss';

import App from './App';

ReactDOM.render(
  <BrowserRouter>
    <Route component={App} />
  </BrowserRouter>,
  document.getElementById('root'),
);
