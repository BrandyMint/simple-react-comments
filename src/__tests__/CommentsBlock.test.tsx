import { mount } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';
import { commentsData } from '../../preview/data';
import Comment from '../components/Comment';
import CommentsBlock from '../components/CommentsBlock';

test('Snapshots', () => {
  const onSubmitMock = jest.fn();
  const componentLoggedIn = renderer.create(
    <CommentsBlock
      comments={commentsData}
      signinUrl="/signin"
      isLoggedIn
      reactRouter={false}
      onSubmit={onSubmitMock}
    />,
  );
  let tree = componentLoggedIn.toJSON();
  expect(tree).toMatchSnapshot();
  componentLoggedIn.unmount();

  const componentNotLoggedIn = renderer.create(
    <CommentsBlock
      comments={commentsData}
      signinUrl="/signin"
      isLoggedIn={false}
      reactRouter={false}
      onSubmit={onSubmitMock}
    />,
  );
  tree = componentNotLoggedIn.toJSON();
  expect(tree).toMatchSnapshot();
  componentNotLoggedIn.unmount();
});

it('should call onSubmit prop with text of form when submitting it', () => {
  const onSubmitMock = jest.fn();
  const commentsBlock = mount(
    <CommentsBlock
      comments={commentsData}
      signinUrl="/signin"
      isLoggedIn
      reactRouter={false}
      onSubmit={onSubmitMock}
    />,
  );
  const preventDefault = jest.fn();
  const value = 'test value';

  commentsBlock.find('textarea').simulate('change', { target: { value } });
  commentsBlock.find('form').simulate('submit', { preventDefault });

  expect(onSubmitMock).toBeCalledWith(value);
  commentsBlock.unmount();
});

it('should change color of comment text', () => {
  const onSubmitMock = jest.fn();
  const customStyles = {
    comment: base => ({
      ...base,
      color: 'red',
    }),
  };
  const commentsBlock = mount(
    <CommentsBlock
      styles={customStyles}
      comments={commentsData}
      signinUrl="/signin"
      isLoggedIn
      reactRouter={false}
      onSubmit={onSubmitMock}
    />,
  );

  const comments = commentsBlock.find(Comment);

  comments.map(comment => {
    const styles = getComputedStyle(comment.getDOMNode());
    expect(styles.color).toBe('red');
  });
  commentsBlock.unmount();
});
