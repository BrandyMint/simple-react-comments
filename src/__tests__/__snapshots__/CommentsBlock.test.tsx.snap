// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots 1`] = `
<div>
  <div>
    <div
      className="css-0"
    >
      <a
        className=""
        href="#"
      >
        <div
          className="css-0__avatar"
          style={
            Object {
              "backgroundImage": "url(https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438)",
            }
          }
        />
      </a>
      <div
        className="css-0__col-right"
      >
        <a
          className=""
          href="#"
        >
          <div
            className="css-0__name"
          >
            Alexey Ryabov
          </div>
        </a>
        <div
          className="css-0__time"
        >
          2018-6-14
        </div>
        <div
          className="css-0__content"
        >
          &lt;h1&gt;hello whats up&lt;/h1&gt;
        </div>
      </div>
    </div>
    <div
      className="css-0"
    >
      <a
        className=""
        href="/asdf"
      >
        <div
          className="css-0__avatar"
          style={
            Object {
              "backgroundImage": "url(https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438)",
            }
          }
        />
      </a>
      <div
        className="css-0__col-right"
      >
        <a
          className=""
          href="/asdf"
        >
          <div
            className="css-0__name"
          >
            Alexey Ryabov
          </div>
        </a>
        <div
          className="css-0__time"
        >
          2018-6-14
        </div>
        <div
          className="css-0__content"
        >
          reactomments is awesome!
        </div>
      </div>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="css-0"
      id="comment-text"
      name="comment-text"
      onChange={[Function]}
      placeholder="Leave a comment"
      value=""
    />
    <button
      className="css-0"
    >
      Comment
    </button>
  </form>
</div>
`;

exports[`Snapshots 2`] = `
<div>
  <div>
    <div
      className="css-0"
    >
      <a
        className=""
        href="#"
      >
        <div
          className="css-0__avatar"
          style={
            Object {
              "backgroundImage": "url(https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438)",
            }
          }
        />
      </a>
      <div
        className="css-0__col-right"
      >
        <a
          className=""
          href="#"
        >
          <div
            className="css-0__name"
          >
            Alexey Ryabov
          </div>
        </a>
        <div
          className="css-0__time"
        >
          2018-6-14
        </div>
        <div
          className="css-0__content"
        >
          &lt;h1&gt;hello whats up&lt;/h1&gt;
        </div>
      </div>
    </div>
    <div
      className="css-0"
    >
      <a
        className=""
        href="/asdf"
      >
        <div
          className="css-0__avatar"
          style={
            Object {
              "backgroundImage": "url(https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438)",
            }
          }
        />
      </a>
      <div
        className="css-0__col-right"
      >
        <a
          className=""
          href="/asdf"
        >
          <div
            className="css-0__name"
          >
            Alexey Ryabov
          </div>
        </a>
        <div
          className="css-0__time"
        >
          2018-6-14
        </div>
        <div
          className="css-0__content"
        >
          reactomments is awesome!
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "fontSize": "1.15em",
        "marginBottom": "10px",
      }
    }
  >
    You should be signed in to write a comment.
  </div>
  <a
    className="css-0"
    href="/signin"
  >
    Sign In
  </a>
</div>
`;
